sopel/db.py

```diff
@@ -201,12 +204,14 @@ def get_nick_id(self, nick, create=True):
         session = self.ssession()
         slug = nick.lower()
         try:
+            self.nick_id_lock.acquire()
             nickname = session.query(Nicknames) \
                 .filter(Nicknames.slug == slug) \
                 .one_or_none()
 
             if nickname is None:
                 if not create:
+                    self.nick_id_lock.release()
```

Comment:
Leaving this in will cause a `RuntimeError` when the lock is released in `finally`.

Reply:
fixed